lib/expressions.js

```diff
@@ -60,20 +60,87 @@ class OperationExpression {
   interpret(context) {
     const args = this.operands.map((x) => x.interpret(context));
     const operator = OPERATORS[this.operator];
-    if (!operator) throw new Error(`Unknown operator: ${operator}`);
+    if (!operator) throw new Error(`Unknown operator: ${this.operator}`);
     return operator(...args);
   }
 
   toExpression() {
     const list = this.operands.map((x) => x.toExpression());
-    return '(' + list.join(this.operator) + ')';
+    return `(${list.join(this.operator)})`;
   }
 
   toJavaScript() {
-    const parameters = Array.from(this.identifiers);
-    const header = '(' + parameters.join(', ') + ')';
-    return header + ' => ' + this.toExpression();
+    let header = '(';
+    for (const identifier of this.identifiers) {
+      if (header.length > 1) header += ', ';
+      header += identifier;
+    }
+    header += ')';
+    return `${header} => ${this.toExpression()}`;
+  }
```

Comment:
I do not understood moving from `Array.from` to `for..of` solution, please explain.
```suggestion
    let header = '';
    for (const identifier of this.identifiers) {
      if (header) header += ', ';
      header += identifier;
    }
    return `(${header}) => ${this.toExpression()}`;
  }
```

Reply:
Murych suggested to avoid creating an extra array. Just not sure if concatenating strings is as fast as Array.join